app_dart/lib/src/service/luci_build_service.dart

```diff
@@ -281,27 +282,33 @@ class LuciBuildService {
   /// The buildset, user_agent, and github_link tags are applied to match the
   /// original build. The build properties and user data from the original build
   /// are also preserved.
-  Future<Build> rescheduleBuild({
-    required String commitSha,
+  Future<Build> reschedulePresubmitBuild({
```

Comment:
nit: might want to leave this and keep it generic (not specify presubmit) as there has been some work to do that. There is also a cleanup issue for this as well.

Reply:
Huh, I intentionally update it specifically for presubmit as this logic will only by consumed by presubmit builds.
For post submit builds, they are handled in a different way as no check run id is needed.

Reply:
Yeah no problem. Not a big deal as I labeled it a nit. I am fine if you leave it as is.

Reply:
Why do we need different logic for presubmit and postsubmit?